src/Voron/StorageEnvironment.cs

```diff
@@ -469,6 +469,8 @@ public void Dispose()
 
                 OnLogsApplied = null;
 
+                _options.NullifyHandlers();
```

Comment:
To be honest I find this confusing as well. I haven't dug too deep into this just saw that those handlers are holding quite a bunch of resources with them and also notice we didn't dispose of the options even though they are disposables.
When and if you should dispose of them is unclear to me but for sure unregistering handlers is required otherwise you might be running handlers of a disposed of ENV in a context of a new ENV

Reply:
@maximburyak we are just interested in the pagers, not he handlers. The handlers are always to the storage env, so that is going to be invalid anyway